test/ruby/test_string_memory.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: false
+require 'test/unit'
+require 'objspace'
+
+class TestStringMemory < Test::Unit::TestCase
+  def capture_allocations(klass)
+    allocations = []
+
+    GC.start
+    GC.disable
+    generation = GC.count
+
+    ObjectSpace.trace_object_allocations do
```

Comment:
Since you are only measuring the number of allocations, and not doing any assertions on the allocated objects, you can simply compare `GC.stat(:total_allocated_objects)` before and after the block.

Reply:
I tried this but there are some internal allocations which we don't care about - we only care about `String` instances. Is there a way to only get the count of strings allocated?

Reply:
Ah right, might bad I missed that you were only considering strings. Forget what I said.